### ISSUE TITLE
How to check the depth level of dependencies when scanning on directories ?

### ISSUE THREAD

**Issue by BinhTNT** [anchore/syft#2346]
Hello,

I'd like to confirm about how depth of dependencies that Syft supports when scanning on directories. Because we want to check if our SBOM is following the Depth element or not. Is there a way that I can verify what is the depth level of dependencies in the case directories scanning. Also, is there an option in CLI to set it ?

Looking forward for your help.
Thanks,
Binh

**Comment by willmurphyscode** [comment]
> Because we want to check if our SBOM is following the Depth element or not

Can you help me understand what this means? I'm not sure what the "depth element" is.

Is there a particular use case you have in mind?

**Comment by SSJune821** [comment]
I also had questions about depth, so I found this issue.

In my opinion, I tried using syft on a Spring Boot project using Maven. If my project uses packages A and B, syft outputs packages A and B as SBOM.
But what if package A was actually created using package A_1 and package A_2?
I understand that, according to the definition of SBOM, sub-packages of the package being used must be imported (Of course, sub-packages of sub-packages and their sub-packages continue to be added... until there are no more sub-packages.)
However, in this case, it seems that syft does not create the contents of packages A_1 and A_2 into SBOM.

I think this is what it means.
(If not, I would like to answer my curiosity.)

thank you
